protocol-units/settlement/mcr/client/src/send_eth_transaction.rs

```diff
@@ -113,9 +113,12 @@ pub async fn send_transaction<
 					"transaction_receipt.gas_used: {} / estimate_gas: {estimate_gas}",
 					transaction_receipt.gas_used
 				);
-				if transaction_receipt.gas_used == estimate_gas {
-					tracing::warn!("Send commitment Transaction  fail because of insufficient gas, receipt:{transaction_receipt:?} ");
-					estimate_gas += (estimate_gas * 10) / 100;
+				// Some valid Tx can abort cause of insufficient gas without consuming all its gas.
+				// Define a threshold a little less than estimated gas to detect them.
+				let tx_gas_consumption_threshold = estimate_gas - (estimate_gas * 10) / 100;
+				if transaction_receipt.gas_used >= tx_gas_consumption_threshold {
+					tracing::info!("Send commitment Transaction  fail because of insufficient gas, receipt:{transaction_receipt:?} ");
+					estimate_gas += (estimate_gas * 30) / 100;
```

Comment:
There is `ReceiptEnvelope::cumulative_gas_used`, I wonder if that gives the true amount.

Reply:
No clear what is it. It seems to be the cumulative gas use by the client, I suppose. Here, I try to estimate how much gas is needed to execute the Tx. With the submit commitment call, Eth doesn't estimate correctly the gas needed. That why I retry by increasing the gas until it pass.